lib/ddtrace/auto_instrument.rb

```diff
@@ -0,0 +1,48 @@
+require 'ddtrace'
+
+module Datadog
+  # AutoInstrumentation enables all integrations
+  module AutoInstrument
+    EXCLUDED_RAILS_INTEGRATIONS = [:rack, :action_cable, :active_support, :action_pack, :action_view, :active_record].freeze
+    EXCLUDED_TEST_INTEGRATIONS = [:cucumber, :rspec].freeze
+
+    module_function
+
+    def patch_all(excluded_integratons = [])
+      integrations = []
+      Datadog.registry.each do |integration|
+        next if excluded_integratons.include?(integration.name)
+        integrations << integration.name
+      end
+
+      Datadog.configure do |c|
+        c.silence_patching_logs
+        # This will activate auto-instrumentation for Rails
+        integrations.each do |integration_name|
+          c.use integration_name
+        end
+      end
+    end
+
+    if defined?(Rails) && defined?(Rails::VERSION) && defined?(Rails::VERSION::MAJOR) &&
```

Comment:
```suggestion
    if defined?(Rails::VERSION::MAJOR) &&
```
`defined?` is safe to use for nested name look-ups, even if the first parts of the constant don't exist.

Reply:
👍 updated